docs/broken_links.py

```diff
@@ -0,0 +1,58 @@
+import re
+import os
+import argparse
+
+from colorama import Fore, Back, Style
+
+files = list()
+os.chdir("source")
+path = os.getcwd()
+broken_count = 0
+clean_count = 0
+
+parser = argparse.ArgumentParser(description='Check broken links')
+parser.add_argument('--verbose', default=False, help="Verbosity level (1: Show all links)")
+
+args = parser.parse_args()
+
+def shortpath(path):
+    return os.path.relpath(path)
+
+def check_broken_links(file):
+    global broken_count
+    global clean_count
+    with open(file,"r") as f:
+        matches = re.findall(r"\[([^\]]+)\]\((?!http)([^[\)#]+)",f.read())
+        if matches:
+            print(Fore.BLUE, "{file}...".format(file=shortpath(file)), Style.RESET_ALL)
+            for match in matches:
+                if match[1]:
+                    extra = match[2] if len(match)>=3 else ""
+                    link_dest = os.path.realpath(os.path.join(os.path.dirname(file), match[1] +".md"))
+                    if not os.path.exists(link_dest):
+                        print("\t\t", Back.RED, Fore.WHITE, "Broken : {link}{extra}".format(link=shortpath(link_dest),extra=extra), Style.RESET_ALL)
+                        broken_count += 1
+                    else:
+                        if args.verbose:
+                            print(Fore.GREEN, "\t\tClean : {link}{extra}".format(link=shortpath(link_dest), extra=extra), Style.RESET_ALL)
+                            clean_count += 1
+        else:
+            if args.verbose:
+                print(Fore.RED, "{file} has no links".format(file=shortpath(file)), Style.RESET_ALL)
+
+def get_md_files(path):
+    global files
```

Comment:
Same thing here, just pass this into the function.

Reply:
Not part of the docs refactor, just a tool I made for my own workflow.